### DISCUSSION TITLE
Static site generator doesn't generate all files

### DISCUSSION THREAD

**Discussion by glow12121** [datenstrom/community#420] [Report a bug] [answered]
Having another issue with building the site...

After the `build` command, what I have in the `public` folder is:

* media/
* 404.html
* index.html
* robots.txt

That is all - no subfolders and respective content. I had five folders in the `content` folder - and the whole site rendered just fine in the `serve` mode.

I already did `update` before building. No error messages appear. What could be the cause?

Thanks!

**Comment by markseuffert** [comment]
Can you give more details, @thumos333. Which folders do you have in your content folder? What happens if you build a new/unmodified Datenstrom Yellow, do you get subfolders? What operating system do you use, please copy/paste your file `system/extensions/yellow.log`.

**Comment by glow12121** [comment]
I have five folders in `content`:

* `1-home` [`page.md`]
* `sectionone` [`page.md` + 13 more .md files]
* `sectiontwo` [`page.md` + another .md file]
* `sectionthree` [`page.md` + 4 more .md files]
* `shared` [`footer.md` and `page-new-default.md`]

* * *

Using Solus 4 Linux, fully updated.

* * *

`yellow.log` follows:

```
2019-03-21 21:03:21 info Install extension 'Shariff 0.8.3'
2019-03-21 21:03:21 info Install extension 'Minishare 0.8.2'
2019-03-31 21:04:32 info Update extension 'Berlin 0.8.3'
2019-03-31 21:04:32 info Update extension 'Bundle 0.8.4'
2019-03-31 21:04:32 info Update extension 'Command 0.8.4'
2019-03-31 21:04:32 info Update extension 'Edit 0.8.5'
2019-03-31 21:04:32 info Update extension 'English 0.8.4'
2019-03-31 21:04:32 info Update extension 'Markdown 0.8.4'
2019-03-31 21:04:32 info Update extension 'Preview 0.8.3'
2019-03-31 21:04:32 info Update extension 'Russian 0.8.4'
2019-03-31 21:04:32 info Update extension 'Stockholm 0.8.3'
```

* * *

I've tried building a default Yellow site - success.

Initially, I was prompted to `configure StaticUrl in file 'system/settings/system.ini'`, which I set as `StaticUrl: http://www.ya.ru` - just a Russian search engine. I entered `build` again - it worked out fine, and `public/about/index.html` showed up. I also tried updating Yellow and re-building - again, success.

* * *

This is my actual `system.ini`: https://gist.github.com/thumos333/67ff58e1c15201c7afd6be03d08a543c

**Comment by markseuffert** [comment]
Thanks, can you also copy/paste the `system/extensions/yellow.log` from the website that worked?

**Comment by glow12121** [comment]
There is no `system/extensions/yellow.log` for the new site :)

I did `serve` (successful) and `build` (successful) one more time - there is still no `yellow.log`.

**Comment by markseuffert** [comment]
Thanks for the information.

 I can't reproduce the described problems on Ubuntu Linux. As far as I can see, everything builds correctly on my machine and the log file is created when installing a new website. Not sure what's the best way to narrow down the problem.

If you want me to investigate, please contact me and I have a closer look.

**Comment by markseuffert** [comment]
Thanks, I can reproduce the build problem. It looks like the build command didn't generate HTML files for some folders. When the `page.md` has setting `Status: hidden`. This shouldn't happen.

**Comment by markseuffert** [comment, marked as answer]
Should be working now. Can you download [latest core](https://github.com/annaesvensson/yellow-core) and test again?

**Comment by markseuffert** [comment]
The logfile is created, as far as I can see. I installed [Solus 4.0 Budgie](https://getsol.us/download/) on Virtualbox 6.0.4 on my Mac. Then downloaded a new Datenstrom Yellow from GitHub, opened a terminal windows and typed `php yellow.php`. Here's the file `system/extensions/yellow.log`:

```
2019-04-05 19:53:32 info Datenstrom Yellow 0.8.3, PHP 7.3.3, CLI Linux
2019-04-05 19:53:32 info Install language 'English'
2019-04-05 19:53:32 info Install language 'French'
2019-04-05 19:53:32 info Install language 'German'
```

Looking good :smile:

**Comment by glow12121** [comment]
Mark, that was a really nice cooperation - thanks for helping me!

Indeed, with the latest core I got the site `built` without a problem - this is great and the site is now ready and online.

The LOG file is also there - both for the actual site and the `master` package. (The latter reads just the same as you copied.)

Many thanks, again!
